src/popup/components/Modals/QrCodeReader.vue

```diff
@@ -34,137 +31,95 @@
         />
       </div>
     </div>
-    <template #footer>
-      <BtnMain
-        :variant="mobile ? 'secondary' : 'primary'"
-        :extend="!mobile"
-        :text="$t('common.ok')"
-        @click="cancelReading"
-      />
+    <template
+      v-if="IS_MOBILE_APP"
+      #footer
+    >
       <BtnMain
-        v-if="mobile"
         :text="$t('modals.qrCodeReader.settings')"
         @click="openSettings"
       />
     </template>
   </Modal>
 </template>
 
-<script>
+<script lang="ts">
+import {
+  computed,
+  onBeforeUnmount,
+  onMounted,
+  ref,
+  watch,
+  defineComponent,
+} from 'vue';
+import { useRoute } from 'vue-router';
+import { useStore } from 'vuex';
 import { BarcodeScanner } from '@capacitor-community/barcode-scanner';
-import { mapMutations } from 'vuex';
+import { BrowserQRCodeReader, IScannerControls } from '@zxing/browser';
+import { useI18n } from 'vue-i18n';
+
 import { IS_EXTENSION, IS_MOBILE_APP } from '@/constants';
 import { handleUnknownError, openInNewWindow } from '@/utils';
-import { RejectedByUserError } from '@/lib/errors';
+import { NoUserMediaPermissionError, RejectedByUserError } from '@/lib/errors';
 
-import Modal from '../Modal.vue';
-import BtnMain from '../buttons/BtnMain.vue';
-import QrScanIcon from '../../../icons/qr-scan.svg?vue-component';
-import IconBoxed from '../IconBoxed.vue';
+import AnimatedSpinner from '@/icons/animated-spinner.svg?skip-optimize';
+import Modal from '@/popup/components/Modal.vue';
+import BtnMain from '@/popup/components/buttons/BtnMain.vue';
+import IconBoxed from '@/popup/components/IconBoxed.vue';
+import QrScanIcon from '@/icons/qr-scan.svg?vue-component';
 
-export default {
+export default defineComponent({
   components: {
     Modal,
     BtnMain,
     IconBoxed,
+    AnimatedSpinner,
   },
   props: {
     title: { type: String, required: true },
     resolve: { type: Function, required: true },
     reject: { type: Function, required: true },
   },
-  setup() {
-    return {
-      QrScanIcon,
-    };
-  },
-  data: () => ({
-    mobile: IS_MOBILE_APP,
+  setup(props) {
     // allow camera while QRScanner is loading to not show cameraNotAllowed before actual check
-    cameraAllowed: IS_MOBILE_APP,
-    browserReader: null,
-    videoInputDevices: [],
-    headerText: '',
-  }),
-  watch: {
-    async cameraAllowed(value) {
-      if (!value) {
-        this.stopReading();
-        return;
-      }
+    const cameraStatus = ref<PermissionState>(
+      IS_MOBILE_APP ? 'granted' : 'denied',
+    );
+    const browserReader = ref<BrowserQRCodeReader | null>(null);
+    const browserReaderControls = ref<IScannerControls>();
+    const qrCodeVideo = ref<HTMLVideoElement>();
```

Comment:
I think it's a good pattern to add "...El" or "...Element" to the variable name if it refers to and HTML element.